### ISSUE TITLE
sharethrough doesn't set bidder code on zero cpm bids

### ISSUE THREAD

**Issue by kitwestneat** [prebid/Prebid.js#1232]
## Type of issue
Bug

## Description
tldr; the sharethrough adapter doesn't properly set the bidderCode when it gets a zero CPM bid, which causes bidmanager to timeout prematurely

When Sharethrough gets a response with no creatives, it generates an exception at `bid.cpm = bidResponse.creatives[0].cpm;` The catch block calls _handleInvalidBid, which creates a new type 2 bid using bidfactory createBid. 

However, _handleInvalidBid doesn't set the bidderCode on the bid before sending it to addBidResponse, and addBidResponse doesn't seem to check. 

Later on in addBidResponse `const { requestId, start } = getBidderRequest(bid.bidderCode, adUnitCode);` is called and since bid.bidderCode is '', start is null. This causes the request to timeout since timeToRespond = responseTimestamp.

## Steps to reproduce
Get a zero bid from sharethrough.

### Expected results

- if getBidderRequest can't find the request and so start is null, addBidResponse shouldn't set timeToRespond
- either sharethrough should set the bidderCode, or createBid should pull it from the bidRequest

**Comment by kitwestneat** [comment]
I just realized this is related to #1210
